packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -472,6 +271,20 @@ class AnalyzeCommand extends FlutterCommand {
     return 0;
   }
 
+  List<File> _collectDartFiles(Directory dir, List<File> collected, {FileFilter exclude}) {
+    for (FileSystemEntity entity in dir.listSync(recursive: false, followLinks: false)) {
+      if (isDartFile(entity) && (exclude == null || !exclude(entity))) {
+        collected.add(entity);
+      }
+      if (entity is Directory) {
```

Comment:
When traversing down, we'll want to stop if we hit a directory with a file named `.dartignore` in it.